File: exercises/domain-3/block-3.4/starter.py
import os


def validate_workflow(content: str) -> list[str]:
    issues = []
    if "pull_request" not in content: issues.append("Must trigger on pull_request")
    if "ANTHROPIC_API_KEY:" in content and "secrets." not in content:
        issues.append("ANTHROPIC_API_KEY must come from secrets")
    if "ANTHROPIC_API_KEY" not in content: issues.append("Must reference ANTHROPIC_API_KEY")
    if "claude" not in content: issues.append("Must invoke claude command")
    if "-p" not in content and "--print" not in content:
        issues.append("Should use claude -p for non-interactive mode")
    if "timeout-minutes" not in content: issues.append("Should set a timeout")
    return issues


path = ".github/workflows/ai-review.yml"
if not os.path.exists(path):
    print(f"❌ File not found: {path}")
else:
    with open(path) as f:
        content = f.read()
    issues = validate_workflow(content)
    if not issues:
        print("✅ Workflow looks good!")
    else:
        for i in issues: print(f"  ❌ {i}")

File: exercises/domain-3/block-3.4/starter.ts
// Exercise 3.4 — Create .github/workflows/ai-review.yml
// This validates the structure of your workflow file

import fs from "fs"

function validateWorkflow(content: string): string[] {
  const issues: string[] = []
  if (!content.includes("pull_request")) issues.push("Must trigger on pull_request")
  if (content.includes("ANTHROPIC_API_KEY:") && !content.includes("secrets.")) {
    issues.push("ANTHROPIC_API_KEY must come from secrets, not hardcoded")
  }
  if (!content.includes("ANTHROPIC_API_KEY")) issues.push("Must reference ANTHROPIC_API_KEY")
  if (!content.includes("claude")) issues.push("Must invoke claude command")
  if (!content.includes("-p") && !content.includes("--print")) issues.push("Should use claude -p for non-interactive mode")
  if (!content.includes("timeout-minutes")) issues.push("Should set a timeout for the Claude step")
  return issues
}

const workflowPath = ".github/workflows/ai-review.yml"
if (!fs.existsSync(workflowPath)) {
  console.log(`❌ File not found: ${workflowPath}`)
} else {
  const content = fs.readFileSync(workflowPath, "utf8")
  const issues = validateWorkflow(content)
  if (issues.length === 0) {
    console.log("✅ Workflow looks good!")
  } else {
    issues.forEach((i) => console.log(`  ❌ ${i}`))
  }
}
